File: src/FoodSharing.Site/Models/Users/UserInfo.cs
namespace FoodSharing.Site.Models.Users;

public class UserInfo
{
    public Guid Id { get; }
    public String Email { get; }
    public String? FirstName { get; }
    public String? LastName { get; }
    public String? Phone { get; }
    public String? AvatarUrl { get; }
    public DateTime RegistrationDate { get; }

    public UserInfo(
        Guid id, String email, String? firstName, String? lastName,
        String? phone, String? avatarUrl, DateTime registrationDate
    )
    {
        Id = id;
        Email = email;
        FirstName = firstName;
        LastName = lastName;
        Phone = phone;
        AvatarUrl = avatarUrl;
        RegistrationDate = registrationDate;
    }
}

File: src/FoodSharing.Site/ClientApp/src/domain/users/userInfo.ts
export class UserInfo {
    constructor(
        public id: string,
        public email: string,
        public firstName: string,
        public lastName: string,
        public phone: string,
        public avatarUrl: string | null,
        public registrationDate: Date
    ) { }

    public get getFullName() {
        if (this.firstName == null && this.lastName == null) return null;

        return `${this.lastName} ${this.firstName}`
    }
}

export function mapToUserInfo(data: any) {
    return new UserInfo(data.id, data.email, data.firstName, data.lastName, data.phone, data.avatarUrl, new Date(data.registrationDate))
}